### PR TITLE
Convert `/src/rageshake` to TS

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: task
Fixes https://github.com/vector-im/element-web/issues/19129

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://615179f65084f13013b3a5fb--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/rageshake/rageshake.ts [hunk 1]
```diff
@@ -415,15 +416,19 @@ class IndexedDBLogStore {
...
     const query = store.openCursor(keyRange);
     return new Promise((resolve, reject) => {
         const results = [];
         query.onerror = (event) => {
+            // @ts-ignore
             reject(new Error("Query failed: " + event.target.errorCode));
         };
         // collect results
         query.onsuccess = (event) => {
+            // @ts-ignore
```
Comment (gsouquet) [inline src/rageshake/rageshake.ts:431]: How come we have to `@ts-ignore` here?
Wondering whether setting a type on the `event` parameter would mitigate the issue?

**Review by gsouquet — APPROVED** [review]
Otherwise looks good to me!

File: src/rageshake/rageshake.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/rageshake/rageshake.ts:431]: I wanted to do that but this doesn't appear to be typed correctly :( Without this, it screams about `EventTarget` not having the properties it has

Comment (SimonBrandner) [reply src/rageshake/rageshake.ts:431]: Screenshot_20210927_095628

For example ^

Comment (SimonBrandner) [reply src/rageshake/rageshake.ts:431]: The `event` only seems to have the type `Event`

Comment (gsouquet) [reply src/rageshake/rageshake.ts:431]: Right, yeah after reading the IndexedDB spec, it doesn't seem that the `onsuccess` event has a proper interface assigned to it.
People seem to be using `any` to remove the warning, but `@ts-ignore` seems quite appropriate too!

**Pull request merged**